tests/lua-tests/src/ParticleTest/ParticleTest.lua

```diff
@@ -230,7 +230,7 @@ end
 
 local function ParticleReorder_onEnterOrExit(tag)
 	local scheduler = cc.Director:getInstance():getScheduler()
-	if tag == "enter" then
+	if tag == "enterTransitionFinish" then
```

Comment:
PR https://github.com/cocos2d/cocos2d-x/pull/18272, add `onEnterCallback` function, move the test case init block into `onEnterCallback`, and then call `onEnterCallback` when __enter__ event comes.

`*_onEnterOrExit` functions were registered in `onEnterCallback`, so that the condition `tag == "enter"` always be false.